### PR TITLE
feat: add content assistance

### PR DESCRIPTION
**Pull request by ivangsa**
**Description**

This is a preliminar version for the following functionality:

- Add SmartPaste functionality to editor
- Add Snippets to VSCode for Content Assistance

**Related issue(s)**
Implements #132 #133

### CODE REVIEW

**Comment by derberg** [comment]
tests are failing because of issues with dependencies installation, related to package-lock v1 vs. v2 issues

problem with `fsevents` that are needed for `chokidar` that is needed by `mocha`. Maybe we can just remove mocha as we anyway do not have tests?

```
npm info it worked if it ends with ok
npm verb cli [
npm verb cli   '/opt/hostedtoolcache/node/14.20.1/x64/bin/node',
npm verb cli   '/opt/hostedtoolcache/node/14.20.1/x64/bin/npm',
npm verb cli   'install',
npm verb cli   '--loglevel',
npm verb cli   'verbose'
npm verb cli ]
npm info using npm@6.14.17
npm info using node@v14.20.1
npm verb npm-session 0fe29d83cbeee9d4
npm info lifecycle asyncapi-preview@0.3.10~preinstall: asyncapi-preview@0.3.10
npm timing stage:loadCurrentTree Completed in 12ms
npm timing stage:loadIdealTree:cloneCurrentTree Completed in 1ms
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
npm timing stage:loadIdealTree:loadShrinkwrap Completed in [5](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:6)03ms
npm http fetch GET 200 https://registry.npmjs.org/fsevents 108ms
npm http fetch GET 200 https://registry.npmjs.org/fsevents/-/fsevents-2.3.2.tgz 38ms
npm timing stage:rollbackFailedOptional Completed in 1ms
npm timing stage:runTopLevelLifecycles Completed in 10[6](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:7)0ms
npm verb stack Error: Unsupported platform for fsevents@2.3.2: wanted {"name":"fsevents","version":"2.3.2","engines":{"node":"^8.16.0 || ^10.6.0 || >=11.0.0"},"os":["darwin"],"dependencies":{},"optionalDependencies":{},"peerDependenciesMeta":{},"devDependencies":{"node-gyp":"^6.1.0"},"bundleDependencies":false,"peerDependencies":{},"deprecated":false,"_resolved":"https://registry.npmjs.org/fsevents/-/fsevents-2.3.2.tgz","_integrity":"sha512-xiqMQR4xAeHTuB9uWm+fFRcIOgKBMiOBP+eXiyT[7](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:8)jsgVCq1bkVygt00oASowB7EdtpOHaaPgKt[8](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:9)12P[9](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:10)ab+DDKA==","_shasum":"8a526f78b8fdf4623b709e0b975c52c24c02fd1a","_shrinkwrap":null,"_id":"fsevents@2.3.2","_from":"fsevents@~2.3.2","_requested":{"type":"range","registry":true,"raw":"fsevents@~2.3.2","name":"fsevents","escapedName":"fsevents","rawSpec":"~2.3.2","saveSpec":null,"fetchSpec":"~2.3.2"},"_spec":"fsevents@~2.3.2","_where":"/home/runner/work/vs-asyncapi-preview/vs-asyncapi-preview/node_modules/chokidar"} (current: {"os":"linux","cpu":"x64"})
npm verb stack     at checkPlatform (/opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/node_modules/npm-install-checks/index.js:46:14)
npm verb stack     at thenWarnEngineIssues (/opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/lib/install/validate-args.js:63:5)
npm verb stack     at /opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/node_modules/iferr/iferr.js:3:64
npm verb stack     at checkEngine (/opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/node_modules/npm-install-checks/index.js:25:[10](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:11))
npm verb stack     at module.exports.isInstallable (/opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/lib/install/validate-args.js:60:3)
npm verb stack     at resolveWithNewModule (/opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/lib/install/deps.js:677:10)
npm verb stack     at /opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/lib/install/deps.js:600:7
npm verb stack     at /opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/node_modules/iferr/iferr.js:3:64
npm verb stack     at /opt/hostedtoolcache/node/14.20.1/x64/lib/node_modules/npm/node_modules/call-limit/call-limit.js:28:43
npm verb stack     at processTicksAndRejections (internal/process/task_queues.js:77:[11](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:12))
npm verb pkgid fsevents@2.3.2
npm verb cwd /home/runner/work/vs-asyncapi-preview/vs-asyncapi-preview
npm verb Linux 5.15.0-1020-azure
npm verb argv "/opt/hostedtoolcache/node/[14](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:15).20.1/x64/bin/node" "/opt/hostedtoolcache/node/14.20.1/x64/bin/npm" "install" "--loglevel" "verbose"
npm verb node v14.20.1
npm verb npm  v6.14.[17](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:18)
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64
npm verb exit [ 1, true ]
npm timing npm Completed in 2826ms

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2022-10-[18](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:19)T11_50_[23](https://github.com/asyncapi/vs-asyncapi-preview/actions/runs/3273141816/jobs/5384988824#step:7:24)_562Z-debug.log
```

File: README.md [hunk 1]
```diff
@@ -14,6 +14,27 @@ You can open AsyncAPI Preview from the editor title/context menu.
 
 Automatic hot-reloading on editor save, but currently, it doesn't reload when saving referenced external files.
 
+## Content Assistance
+
+### Available snippets:
+
+Open an empty or otherwise yaml file and start typing one of the following prefixes, you may need to press `Ctrl+space` to trigger autocompletion in some cases:
```
[the remark this answers is not in the document]
Comment (ivangsa) [reply README.md:21]: Yes, this content assistance it's only available on ymls

**Comment by ivangsa** [comment]
for some reason just adding `genson-js` which is in charge of creating a jsonschema from a json example, messed with the fsevents dependency :/

**Review by derberg — APPROVED** [review]

**Pull request merged**